### DISCUSSION TITLE
eml file format

### DISCUSSION THREAD

**Discussion by justin2004** [saulpw/visidata#1261] [General]
Hello,
When you open a .eml file it does not seem possible to look at the actual content.
For example [this .eml](https://www.phpclasses.org/browse/download/1/file/14672/name/message.eml) file has some plan text in it but it looks like Visidata can only show that it has plan text.

Is this correct or I am not understanding how to navigate to the actual content?
 

|filename        |content\_type          |payload  |
|----------------|-----------------------|--------:|
|                |multipart/mixed        |2        |
|                |multipart/related      |3        |
|                |multipart/alternative  |2        |
|                |text/plain             |87       |
|                |text/html              |962      |
|logo\.gif       |image/gif              |1617     |
|background\.gif |image/gif              |4414     |
|attachment\.txt |text/plain             |90       |

**Comment by daviewales** [comment]
It looks like that's how the loader currently works.
It should be possible to improve the loader to allow you to dive into rows to see the contents of the sections.
See the loader code here:
https://github.com/saulpw/visidata/blob/stable/visidata/loaders/eml.py

**Reply by saulpw** [reply]
Thanks @daviewales!  Yes, this is exactly correct.  If someone does improve the loader (shouldn't be too hard), please submit a PR!
